pytest_cases/tests/pytest_extension/parametrize_plus/test_filter.py

```diff
@@ -0,0 +1,67 @@
+from pytest_cases import filters, case, parametrize_with_cases
+
+A = "a"
+B = "b"
+C = "c"
+
+
+@case(tags=[A])
+def case_one():
+    return 1
+
+
+@case(tags=[A, B])
+def case_two():
+    return 2
+
+
+@case(tags=[B, C])
+def case_three():
+    return 3
+
+
+@case(tags=[A, C])
+def case_four():
+    return 4
+
+
+@parametrize_with_cases(
+    argnames="value", cases=".", filter=filters.has_tag(B)
+)
+def test_filter_with_tag(value):
+    print(value)
+
+
+@parametrize_with_cases(
+    argnames="value", cases=".", filter=~filters.has_tag(B)
+)
+def test_filter_without_tag(value):
+    print(value)
+
+
+@parametrize_with_cases(
+    argnames="value", cases=".", filter=filters.has_tag(B) & filters.has_tag(C)
+)
+def test_filter_with_and_relation(value):
+    print(value)
+
+
+@parametrize_with_cases(
+    argnames="value", cases=".", filter=filters.has_tag(B) | filters.has_tag(C)
+)
+def test_filter_with_or_relation(value):
+    print(value)
+
+
+@parametrize_with_cases(
+    argnames="value", cases=".", filter=filters.has_prefix("case_t")
```

Comment:
See above: if you had written "data_t" here you would not be able to retrieve anythin, because `prefix="case_"` and is applied before filters are executed. We should change this to `has_id_prefix` as suggested above.